test/11_holograph_factory_tests.ts

```diff
@@ -1,61 +1,242 @@
-describe('Holograph Factory Contract', async function () {
-  describe('init():', async function () {
-    it('should successfully init once'); // Validate hardcoded values are correct
-    it('should fail to init if already initialized');
-    it('should fail if data parameter is invalid '); // NOTE: check if the expected value is correct
-  });
+import { PreTest } from './utils';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import setup from './utils';
+import { Signature, StrictECDSA, generateInitCode, generateErc20Config } from '../scripts/utils/helpers';
+import { ONLY_ADMIN_ERROR_MSG } from './utils/error_constants';
+import { ConfigureEvents } from '../scripts/utils/events';
+
+describe('Holograph Factory Contract', async () => {
+  let l1: PreTest;
+
+  let Mock: any;
+  let mock: any;
+  let accounts: SignerWithAddress[];
+  let deployer: SignerWithAddress;
+  let owner: SignerWithAddress;
+  let nonOwner: SignerWithAddress;
+  let mockSigner: SignerWithAddress;
+  let chainId: number;
+
+  const randomAddress = () => ethers.Wallet.createRandom().address;
+
+  before(async () => {
+    l1 = await setup();
+    accounts = await ethers.getSigners();
+    deployer = accounts[0];
+    owner = accounts[1];
+    nonOwner = accounts[2];
 
-  describe(`bridgeIn()`, async function () {
-    it('should return the expected selector from the input payload');
-    it('should return bad data if payload data is invalid');
+    chainId = (await ethers.provider.getNetwork()).chainId;
+
+    Mock = await ethers.getContractFactory('Mock');
+    mock = await Mock.deploy();
+    await mock.deployed();
+
+    mockSigner = await ethers.getSigner(mock.address);
   });
 
-  describe('bridgeOut()', async function () {
-    it('should return selector and payload');
+  describe('init():', async () => {
+    // TODO: Check initialized
+    // it.only('should check that contract was successfully initialized once', async () => {
+    //   await expect(l1.holographFactory.connect(deployer)._isInitialized().to.equal(true);
+    // });
+
+    it('should fail if already initialized', async () => {
+      const initCode = generateInitCode(
+        ['address', 'address'],
+        [l1.holographFactory.address, l1.holographRegistry.address]
+      );
+      await expect(l1.holographFactory.connect(deployer).init(initCode)).to.be.reverted;
+    });
   });
 
-  describe('deployHolographableContract()', async function () {
-    it('should fail with invalid signature if config is incorrect');
-    it('should fail with invalid signature if signature.r is incorrect');
-    it('should fail with invalid signature if signature.s is incorrect');
-    it('should fail with invalid signature if signature.v is incorrect');
-    it('should fail with invalid signature if signer is incorrect');
+  describe('deployHolographableContract()', async () => {
```

Comment:
I just realized that this function does not have a success case test. Please add one.